### PR TITLE
[frontend/backend] Fix incorrect use of stixCoreRelationshipDistribution queries (#6873)

### LINKED ISSUE #6873: [technical] incorrect use of stixCoreRelationshipDistribution queries
In Report details, the entities distribution uses widget `EntityStixCoreRelationshipsHorizontalBars` which uses the query `stixCoreRelationshipsDistribution`.

In this case we send `relationship_type: "object"` to this query, but this is a ref relationship not a stix-core-relationship.

With #6872 we now properly check if the provided `relationship_type` is consistent with the query expectations, so sending `object` would throw an error.

We currently disabled the error throwing in the check in stixCoreRelationshipsDistribution to give time to fix the issue frontend side.

We need to properly use other widget such as `StixRelationshipsHorizontalBars` instead of `EntityStixCoreRelationshipsHorizontalBars`, but this means a refactoring (StixRelationshipsHorizontalBars currently won't accept props such as `relationship_type`).

We also need to check throughout the frontend code if there are other places with failing widgets for same reasons.

### PR DESCRIPTION
**Pull request by marieflorescontact**
### Proposed changes

* use `StixRelationshipsHorizontalBars` instead of ` EntityStixCoreRelationshipsHorizontalBars` in `GroupingDetails` and `reportDetails` components => these were the only 2 components where  `EntityStixCoreRelationshipsHorizontalBars` used an `object` relationshipType
*  error  throws for stixCoreRelationshipDistribution query is not commented any more.

### Related issues

* [issue/6873](https://github.com/OpenCTI-Platform/opencti/issues/6873)

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/analyses/reports/ReportDetails.jsx [hunk 1]
```diff
@@ -164,15 +199,17 @@ const ReportDetailsComponent = (props) => {
...
               fromId={report.id}
-              toTypes={['Stix-Core-Object']}
+              startDate={null}
+              endDate={null}
               relationshipType="object"
-              field="entity_type"
-              seriesName={t('Number of entities')}
-              isTo={true}
+              dataSelection={entitiesDidstributionDataSelection}
+              parameters={'Entities distribution'}
```
Comment (labo-flg) [inline opencti-platform/opencti-front/src/private/components/analyses/reports/ReportDetails.jsx]: parameters is not an object?

Comment (labo-flg) [reply opencti-platform/opencti-front/src/private/components/analyses/reports/ReportDetails.jsx]: judging from the use of `parameters` [here](https://github.com/OpenCTI-Platform/opencti/blob/c394cc812d99db764aba933376cd0089990eba1a/opencti-platform/opencti-front/src/private/components/common/stix_relationships/StixRelationshipsHorizontalBars.jsx#L166) and [there](https://github.com/OpenCTI-Platform/opencti/blob/c394cc812d99db764aba933376cd0089990eba1a/opencti-platform/opencti-front/src/private/components/common/stix_relationships/StixRelationshipsHorizontalBars.jsx#L184) you probably
want to set `parameters={{ title: 'Entities distribution' }}`

File: opencti-platform/opencti-front/src/private/components/analyses/reports/ReportDetails.jsx [hunk 2]
```diff
@@ -122,6 +122,41 @@ const ReportDetailsComponent = (props) => {
...
+      dynamicFrom: {
+        mode: 'and',
+        filters: [],
+        filterGroups: [],
+      },
+      dynamicTo: {
+        mode: 'and',
+        filters: [],
+        filterGroups: [],
+      },
```
Comment (labo-flg) [inline opencti-platform/opencti-front/src/private/components/analyses/reports/ReportDetails.jsx]: you can `import { emptyFilterGroup } from '../../../utils/filters/filtersUtils';` to simplify this part

```suggestion
      dynamicFrom: emptyFilterGroup,
      dynamicTo: emptyFilterGroup,
```

**Comment by marieflorescontact** [comment]
PR status: all comments have been taken into account

File: opencti-platform/opencti-front/src/private/components/analyses/reports/ReportDetails.jsx [hunk 3]
```diff
@@ -164,15 +192,17 @@ const ReportDetailsComponent = (props) => {
...
+              startDate={null}
+              endDate={null}
               relationshipType="object"
-              field="entity_type"
-              seriesName={t('Number of entities')}
-              isTo={true}
+              dataSelection={entitiesDistributionDataSelection}
+              parameters={{ title: 'Entities distribution' }}
+              variant="inEntity"
+              withExportPopover={true}
```
Comment (labo-flg) [inline opencti-platform/opencti-front/src/private/components/analyses/reports/ReportDetails.jsx]:
```suggestion
```

File: opencti-platform/opencti-front/src/private/components/analyses/groupings/GroupingDetails.jsx [hunk 4]
```diff
@@ -133,15 +161,17 @@ const GroupingDetailsComponent = (props) => {
...
+              startDate={null}
+              endDate={null}
               relationshipType="object"
-              field="entity_type"
-              seriesName={t('Number of entities')}
-              isTo={true}
+              dataSelection={entitiesDistributionDataSelection}
+              parameters={{ title: 'Entities distribution' }}
+              variant="inEntity"
+              withExportPopover={true}
```
Comment (labo-flg) [inline opencti-platform/opencti-front/src/private/components/analyses/groupings/GroupingDetails.jsx]:
```suggestion
```

**Review by labo-flg — CHANGES_REQUESTED** [review]
We do not want to be able to export as image the result of this widget, if we want same behavior as before

File: opencti-platform/opencti-front/src/private/components/analyses/reports/ReportDetails.jsx (hunk 3 above)
Comment (marieflorescontact) [reply opencti-platform/opencti-front/src/private/components/analyses/reports/ReportDetails.jsx]: oups I forgot this

**Review by labo-flg — APPROVED** [review]

**Pull request merged**